lib/spack/spack/spec.py

```diff
@@ -511,8 +511,17 @@ def __init__(self, *args):
             raise TypeError(
                 "__init__ takes 1 or 2 arguments. (%d given)" % nargs)
 
-    def _add_version(self, version):
-        self.versions.add(version)
+    def _add_versions(self, version_list):
+        # If it already has a non-trivial version list, this is an error
+        if self.versions and self.versions != vn.VersionList(':'):
+            # Note: This may be impossible to reach by the current parser
+            # Keeping it in case the implementation changes.
+            raise MultipleVersionError(
+                'A spec cannot contain multiple version signifiers.'
+                ' Use a version list instead.')
+        self.versions = vn.VersionList()
+        for version in version_list:
+            self.versions.add(version)
```

Comment:
@alalazo I thought about that... I don't love the idea of having a function that's not a class method changing the fields of a class, but it would allow us to reuse the code.

I gather from the question you think that's preferable to this?

Reply:
Well, I have the same doubt. I think it's fine as long as it is "private" i.e. starting with an underscore.

Reply:
I think this should eventually be implemented as a class/mixin "VersionedSpecType" that factors out all of the common code for versioned items. But I think that is out of scope for this PR.